src/host/hardware_timer/timer.c

```diff
@@ -13,21 +13,31 @@
 #endif
 
 // in our case not a busy wait
-PICO_WEAK_FUNCTION_DEF(busy_wait_us)
+PICO_WEAK_FUNCTION_DEF(busy_wait_us_32)
 void PICO_WEAK_FUNCTION_IMPL_NAME(busy_wait_us_32)(uint32_t delay_us) {
 #if defined(__unix__) || defined(__APPLE__)
     usleep(delay_us);
 #else
     assert(false);
 #endif
 }
+
 PICO_WEAK_FUNCTION_DEF(busy_wait_us)
 void PICO_WEAK_FUNCTION_IMPL_NAME(busy_wait_us)(uint64_t delay_us) {
     absolute_time_t t;
     update_us_since_boot(&t, time_us_64() + delay_us);
     busy_wait_until(t);
 }
 
+PICO_WEAK_FUNCTION_DEF(busy_wait_ms)
+void PICO_WEAK_FUNCTION_IMPL_NAME(busy_wait_ms)(uint32_t delay_ms) {
+#if defined(__unix__) || defined(__APPLE__)
+    usleep(delay_ms * 1000);
```

Comment:
https://man7.org/linux/man-pages/man3/usleep.3.html says that `usleep` fails if "usec is greater than or equal to 1000000", which I guess means this `busy_wait_ms` function will fail if passed a `delay_ms` value >= 1000 ?
Is that something we ought to check for, or are we not really that concerned about the correct operation of "host platform mode"? :man_shrugging: